### PR TITLE
Add retry on generating CreateSnapshot task in case volume itself is NotFound

### PR DESCRIPTION
**Pull request by lintongj**
The following changes are made in this pull request.
- Add retry on NotFound error when generating CreateSnapshot task.
- Update e2e test case for verifying the change. 

Prerequisite: Enabling certain feature switch on VCenter.  

Testing: **Note**: Please replace `<vc user>`, `<vc password>` and  `<vc-ip>` with yours.

- Command: `ASTROLABE_VC_URL='https://<vc user>:<vc password>@<vc-ip>/sdk' ENABLE_DEBUG_LOG=1 go test -v -run TestBackupIVDs`

- Result: 
  - Before adding the retry logic, the test case would fail at CreateSnapshot API call for NotFound error.
  - After adding the retry logic, the test case would pass.

### CODE REVIEW

File: pkg/ivd/ivd_protected_entity.go [hunk 1]
```diff
@@ -295,13 +295,20 @@ func (this IVDProtectedEntity) Snapshot(ctx context.Context, params map[string]m
...
 		this.logger.Infof("Retrying CreateSnapshot on IVD Protected Entity, %v, for one hour at the maximum, Current retry count: %d", this.GetID().String(), retryCount)
+		retryCount++
 		var vslmTask *vslm.Task
 		vslmTask, err := this.ipetm.vslmManager.CreateSnapshot(ctx, NewVimIDFromPEID(this.GetID()), "AstrolabeSnapshot")
 		if err != nil {
+			if soap.IsSoapFault(err) {
+				fault := soap.ToSoapFault(err).Detail.Fault
+				if _, ok := fault.(vim.NotFound); ok {
+					this.logger.WithError(err).Errorf("Creating a task for the CreateSnapshot request failed with NotFound. Will retry in %v second(s)", retryInterval)
+					return false, nil
```
Comment (deepakkinni) [inline pkg/ivd/ivd_protected_entity.go:306]: We return false for all errors, which triggers the retry anyways(see line 309). Is there a need to check error type?

Comment (lintongj) [reply pkg/ivd/ivd_protected_entity.go:306]: Actually, not. 

`wait.PollImmediate` only retries on `false, nil` not `false, <some error>`

**Review by deepakkinni — APPROVED** [review]
LGTM

File: pkg/ivd/ivd_protected_entity.go [hunk 2]
```diff
@@ -295,13 +295,20 @@ func (this IVDProtectedEntity) Snapshot(ctx context.Context, params map[string]m
 	retrieveSnapDetailsErr := 0
 	err := wait.PollImmediate(retryInterval, time.Hour, func() (bool, error) {
 		this.logger.Infof("Retrying CreateSnapshot on IVD Protected Entity, %v, for one hour at the maximum, Current retry count: %d", this.GetID().String(), retryCount)
+		retryCount++
 		var vslmTask *vslm.Task
 		vslmTask, err := this.ipetm.vslmManager.CreateSnapshot(ctx, NewVimIDFromPEID(this.GetID()), "AstrolabeSnapshot")
 		if err != nil {
+			if soap.IsSoapFault(err) {
+				fault := soap.ToSoapFault(err).Detail.Fault
+				if _, ok := fault.(vim.NotFound); ok {
```
Comment (xing-yang) [inline pkg/ivd/ivd_protected_entity.go:304]: Could this "NotFound" be caused by something else other than FCD not found?
Can you add a comment here?

**Pull request closed without merging**